# Statistical Analysis

### Possible Research Qs
- Does sleep affect mental health?
- Does diet affect mental health?
- Does financial stress affect mental health?
- Do depressed students perform worse/better academically? (gpa)

In [1]:
# install.packages("dplyr")
library(dplyr)


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




In [2]:
df = read.csv('data/clean_dataset.csv')
df = df[,-1]
head(df, 15)

id  gender age academic_pressure work_pressure gpa  study_sat
1    2 Male   33  5                 0             8.97 2        
2    8 Female 24  2                 0             5.90 5        
3   26 Male   31  3                 0             7.03 5        
4   30 Female 28  3                 0             5.59 2        
5   32 Female 25  4                 0             8.13 3        
6   33 Male   29  2                 0             5.70 3        
7   52 Male   30  3                 0             9.54 4        
8   56 Female 30  2                 0             8.04 4        
9   59 Male   28  3                 0             9.79 1        
10  62 Male   31  2                 0             8.38 3        
11  83 Male   24  3                 0             6.10 3        
12  91 Male   33  3                 0             7.03 4        
13  94 Male   27  5                 0             7.04 1        
14 100 Female 19  2                 0             8.52 4        
15 103 Female 19  5                 0             5.64 5        
   sleep_dur           diet      suicidal_thoughts study_hrs financial_stress
1  '5-6 hours'         Healthy   Yes                3        1               
2  '5-6 hours'         Moderate  No                 3        2               
3  'Less than 5 hours' Healthy   No                 9        1               
4  '7-8 hours'         Moderate  Yes                4        5               
5  '5-6 hours'         Moderate  Yes                1        1               
6  'Less than 5 hours' Healthy   No                 4        1               
7  '7-8 hours'         Healthy   No                 1        2               
8  'Less than 5 hours' Unhealthy No                 0        1               
9  '7-8 hours'         Moderate  Yes               12        3               
10 'Less than 5 hours' Moderate  Yes                2        5               
11 '5-6 hours'         Moderate  Yes               11        1               
12 'Less than 5 hours' Healthy   Yes               10        2               
13 'Less than 5 hours' Moderate  No                10        1               
14 'Less than 5 hours' Unhealthy No                 6        2               
15 'Less than 5 hours' Moderate  Yes                4        5               
   family_hist depression age_lab sleep_lab
1  No          1          31+     >5       
2  Yes         0          23-26   >5       
3  Yes         0          31+     <5       
4  Yes         1          27-30   >5       
5  No          0          23-26   >5       
6  No          0          27-30   <5       
7  No          0          27-30   >5       
8  Yes         0          27-30   <5       
9  No          1          27-30   >5       
10 No          1          31+     <5       
11 Yes         1          23-26   >5       
12 Yes         0          31+     <5       
13 Yes         1          27-30   <5       
14 Yes         0          18-22   <5       
15 Yes         1          18-22   <5

## Does Lack of Sleep Affect Depression?

Investigate whether getting less than 5 hours of sleep is associated with a higher rate of depression ($\bar{x}_1$) than that of students who sleep more than 5 hours per night ($\bar{x}_2$). Use significance level $\alpha = 0.01$.

$H_0$: $\bar{x}_1 - \bar{x}_2 = 0$ <br>
$H_A$: $\bar{x}_1 - \bar{x}_2 > 0$ ($\bar{x}_1 > \bar{x}_2$)

In [10]:
# Group dataframe
sleep <- df %>%
    group_by(sleep_lab) %>%
    summarise(
        sum = sum(depression),
        count = n(),
        proportion = sum/count)
sleep

sleep_lab sum   count proportion
1 <5         5352  8297 0.6450524 
2 >5        10938 19540 0.5597748

In [11]:
# Prop.test
counts = sleep$sum
n = sleep$count
result = prop.test(counts, n, alternative='greater', conf.level=.99)
print(result)
p_value = result$p.value
p_value


	2-sample test for equality of proportions with continuity correction

data:  counts out of n
X-squared = 174.13, df = 1, p-value < 2.2e-16
alternative hypothesis: greater
99 percent confidence interval:
 0.07044064 1.00000000
sample estimates:
   prop 1    prop 2 
0.6450524 0.5597748 



[1] 4.636815e-40

### Interpretation

p-value = $4.637$ x $10^{-40}$ < $0.01 = \alpha$

Therefore, there is evidence at the $99$% confidence level that students who get less than 5 hours of sleep per night have a higher rate of depression than students who get more than 5 hours of sleep per night. We reject the null hypothesis.

## Research Question 2

## Research Question 3